index.rst

```diff
@@ -1,61 +1,546 @@
+configurations with different labels. Configurable CSCs must specify a list of
+recommended labels. How they implement this is up to the CSC. It should be
+noted that *not all configuration labels need to be related with a recommended
+configuration*. For instance, old configurations that are still valid can be
+kept with the repository without a label. This will allow knowledgeable users
+to use them if needed.
+
+At minimum, all configurable CSCs should pass at least one label in the
+``recommendedSettingsLabels`` attribute, which can be explicitly referenced in
+the ``settingsToApply`` attribute of the ``start`` command. The CSC should
+understand how to use this label to retrieve the correct configuration. See
+caveats to this process below.
+
+Settings Applied
+^^^^^^^^^^^^^^^^
+
+The ``settingsApplied`` event is a Generic Event that is to be implemented by
+every CSC.  It currently contains two parameters: ``settingsVersion`` and
+``otherSettingsEvent``.  This event should be published between after the
+``start`` command starts and before it finishes.
+
+The ``settingsVersion`` will contain the SHA or name corresponding to the
+published configuration.
+
+The ``otherSettingsEvents`` is a comma-separated list of other specific CSC
+configuration events. This may be blank if no other specific CSC events are
+necessary. If ``otherSettingsEvents`` is not blank, then those event(s) must
+be published by the CSC alongside the ``settingsApplied`` event. The CSC is
+allowed to publish as many events as necessary to convey the information.
+
+Other Settings Applied events
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Since it is not possible to provide a generic way for CSCs to output detailed
+information about the configuration parameters they are loading, it is
+recommended to create additional events (particular to each CSC) to carry
+that information.
+
+Although it is not required, for clarity, we suggest that these events be
+preceded by ``settingsApplied`` followed by some description of the content,
+e.g., ``settingsAppliedLUT`` or ``settingsAppliedController``.
+
+Available solutions and frameworks
+==================================
+
+Salobj Derived CSCs
+-------------------
+
+`Salobj <https://ts-salobj.lsst.io>`__ is the framework provided by Telescope
+& Site to develop CSCs in python. Extensive development documentation is
+available, especially on how to create
+`configurable CSCs <https://ts-salobj.lsst.io/salobj_cscs.html#writing-a-csc>`__.
+
+Components that are written using the framework will automatically inherit the
+standard behaviour implemented in the library. The main points regarding
+Salobj CSCs are:
+
+  #. Definition of the configuration repo. In general CSC configuration should
+     be grouped according to the overall system architecture. For instance,
+     `ts_config_attcs <https://github.com/lsst-ts/ts_config_attcs>`__ hosts
+     configurations for all the `ATCS` configurable components.
```

Comment:
So, we got _attcs_ and _ATCS_ and I know we are trying to keep from repeating telescope twice, but we should pick one and stick with it.

Reply:
That's a good point... we may have to resolve this later though as it will either involve changing the repository name or code in some repositories.